File: joonas/075.cpp
class Solution {
public:
    void rotate(vector<vector<int>>& m) {
        int n = m.size();
        for(int j=0; 2 * j < n; ++j) {
            for(int i=0; i < n - 2 * j - 1; ++i) {
                int t = m[j][i+j];
                m[j][i+j] = m[n-1-i-j][j];
                m[n-1-i-j][j] = m[n-1-j][n-1-i-j];
                m[n-1-j][n-1-i-j] = m[i+j][n-1-j];
                m[i+j][n-1-j] = t;
            }
        }
    }
};

File: jlogkim/075.java
class Solution {
    public void rotate(int[][] matrix) {
        //(r,c) -> (c, n-1-r)
        int n = matrix.length;
        int tmp;
        for(int r=0;r<n/2;r++){
            for(int c=r;c<(n-1-r);c++){
                tmp=matrix[n-1-c][r];
                matrix[n-1-c][r]=matrix[n-1-r][n-1-c];
                matrix[n-1-r][n-1-c]=matrix[c][n-1-r];
                matrix[c][n-1-r]=matrix[r][c];
                matrix[r][c]=tmp;
            }
        }
    }
}